storage/include/storage/storage_metrics.h

```diff
@@ -14,41 +15,67 @@
 namespace concord {
 namespace storage {
 
+class StorageMetrics {
+ public:
+  virtual void updateMetrics() = 0;
+};
+
+class StorageMetricsUpdator {
```

Comment:
I think this looks good and does the job. I would suggest to use the constructor and destructor to start/stop, without explicit start() and close() calls. That would allow us to not implement the destructors of the metrics objects (InMemoryStorageMetrics and RocksDbStorageMetrics). Additionally, users won't be able to start multiple times and we don' t need to guard against that case.

I was thinking another option might be to have an utility class that runs a function every X seconds, like so:

```
class PeriodicCall {
 public:
   PeriodicCall(const std::function<void()>& f, const std::chrono::milliseconds &interval);
   ~PeriodicCall(); // sets the stop flag and calls t.join()

 private:
  std::thread t;
  std::chrono::milliseconds interval;
  std::atomic_bool stop{false};
};
```
We can then use PeriodicCall as a member in the metrics classes (as now), but avoid inheritance, i.e. remove the `StorageMetrics` interface. For example:

```
InMemoryStorageMetrics::InMemoryStorageMetrics() :
  ... 
  periodic_call_{[this] () { updateMetrics(); }, metrics_update_interval_},
  ...
```

Reply:
Actually, we want to be calling `updateMetrics()` after the constructor has done its job. Is that the reason why you have an explicit `start()` method? I think we can still use constructors by having a default one that doesn't start the thread and using the default move constructor, something like:
```
metrics_updator_ = StorageMetricsUpdator{this};
```

or

```
// Not setting it in the initialiser list.
periodic_call_ = PeriodicCall{[this] () { updateMetrics(); }, metrics_update_interval_};
```